splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/OrderByIT.java

```diff
@@ -377,4 +382,48 @@ public void testOrderByClauseWithNullInWindowFunctionThroughSparkPath() throws E
         rs.close();
     }
 
+    @Test
+    public void testAliasInOrderComplex() throws Exception {
+        String sqlText = "select sum(a1) as ALIAS from t2 order by -ALIAS";
+        String expected = "ALIAS |\n" +
+                         "--------\n" +
+                         "  15   |" ;
+        SqlExpectToString( methodWatcher, sqlText, expected, false );
+
+        sqlText = "select sqrt(a1*a1)*2 as ALIAS from t2 order by -2*ALIAS+a1";
+        expected = "ALIAS |\n" +
+                  "--------\n" +
+                  " 10.0  |\n" +
+                  "  8.0  |\n" +
+                  "  6.0  |\n" +
+                  "  4.0  |\n" +
+                  "  2.0  |";
+        SqlExpectToString( methodWatcher, sqlText, expected, false );
+    }
+
+    @Test
+    public void testAliasInOrderSimple() throws Exception {
+        // use these TWO order by to make sure ResultColumnList correctly calculates getFirstOrderByIndex.
+        String sqlText = "select a1 as ALIAS from t2 order by -ALIAS, -2*ALIAS";
+        String expected = "ALIAS |\n--------\n   5   |\n   4   |\n   3   |\n   2   |\n   1   |";
+        SqlExpectToString( methodWatcher, sqlText, expected, false );
+    }
+
+    // make sure we disable access to alias for WHERE, HAVING and GROUP BY
+    @Test
+    public void testUsingAliasOnlyInOrderBy() throws Exception {
+        String sqlTexts[] =
+                {       "select a1+1 as ALIAS from t2 WHERE ALIAS > 0 ORDER BY -ALIAS",
+                        "select a1+1 as ALIAS from t2 GROUP BY ALIAS ORDER BY -ALIAS",
+                        "select a1, sum(b1+1) as ALIAS from t1 GROUP BY a1 HAVING ALIAS > 0 ORDER BY -ALIAS"
+                        };
+        for( String sqlText : sqlTexts ) {
+            SqlExpectException( methodWatcher, sqlText, "42X04" );
+        }
+    }
```

Comment:
As the comment says, this makes sure we disable access to alias for WHERE, HAVING and GROUP BY.
Currently, this is not supported. If we remove the line with [clearAliases](https://github.com/splicemachine/spliceengine/pull/3497/files#diff-516cd97dea19fe70597a7399aec1534cR1424) we can enable this. @yxia92 and i agreed on it might be safe currently to not change the behavior here.